src/common/line_entry.rs

```diff
@@ -271,4 +318,104 @@ mod tests {
             assert!(comment.is_none());
         }
     }
+
+    mod get_substitution_keys {
+        use super::*;
+
+        #[test]
+        fn run_with_empty() {
+            let input = line_entry(1, 1, "");
+            assert!(input.get_substitution_keys().is_empty());
+        }
+
+        #[test]
+        fn run_with_simple() {
+            let input = line_entry(1, 1, "FOO=$BAR");
+            assert_eq!(input.get_substitution_keys(), vec!["BAR"]);
+        }
+
+        #[test]
+        fn run_with_simple_comment() {
+            let input = line_entry(1, 1, "FOO=$BAR # comment");
+            assert_eq!(input.get_substitution_keys(), vec!["BAR"]);
+        }
+
+        #[test]
+        fn run_with_curly_braces() {
+            let input = line_entry(1, 1, "FOO=${BAR}");
+            assert_eq!(input.get_substitution_keys(), vec!["BAR"]);
+
+            let input = line_entry(1, 1, "FOO=$BAR}");
+            assert_eq!(input.get_substitution_keys(), vec!["BAR"]);
+
+            let input = line_entry(1, 1, "FOO=${BAR");
+            assert!(input.get_substitution_keys().is_empty());
+        }
+
+        #[test]
+        fn run_with_double_quotes() {
+            let input = line_entry(1, 1, r#"FOO="$BAR""#);
+            assert_eq!(input.get_substitution_keys(), vec!["BAR"]);
+
+            let input = line_entry(1, 1, r#"FOO=$BAR""#);
```

Comment:
```suggestion
            let input = line_entry(1, 1, r#"FOO=$BAR"#);
```
Is that an extra quote?